### PR TITLE
[Merged by Bors] - feat: bialgebra isomorphisms

### PR DESCRIPTION
**Pull request by 101damnations**
Isomorphisms of bialgebras. Mimics `Mathlib.Algebra.Module.Equiv`

---

Not sure about the notation; it's supposed to be a combination of the notation for coalgebra equivs and algebra equivs.

- [x] depends on: #11961 
- [x] depends on: #11962 
- [x] depends on: #11966 
- [x] depends on: #11970 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Bialgebra/Equiv.lean [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+## Main definitions
+
+* `BialgEquiv R A B`: the type of `R`-bialgebra isomorphisms between `A` and `B`.
+
+## Notations
+
+* `A ≃ₐc[R] B` : `R`-bialgebra equivalence from `A` to `B`.
+-/
+
+set_option autoImplicit true
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Bialgebra/Equiv.lean:24]: Do we really want this?

File: Mathlib/RingTheory/Bialgebra/Equiv.lean [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+/-- An equivalence of bialgebras is an invertible bialgebra homomorphism. -/
+structure BialgEquiv (R : Type u) [CommSemiring R] (A : Type v) (B : Type w)
+    [Semiring A] [Semiring B] [Algebra R A] [Algebra R B]
+    [CoalgebraStruct R A] [CoalgebraStruct R B] extends A ≃ₗc[R] B, A ≃* B where
+
+attribute [nolint docBlame] BialgEquiv.toMulEquiv
+attribute [nolint docBlame] BialgEquiv.toCoalgEquiv
+
+@[inherit_doc BialgEquiv]
+notation:50 A " ≃ₐc[" R "] " B => BialgEquiv R A B
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Bialgebra/Equiv.lean:43]: The `c` stands for coalgebra, right?

**Review by riccardobrasca — APPROVED** [review]

File: Mathlib/RingTheory/Bialgebra/Equiv.lean (hunk 2 above)
Comment (101damnations) [reply Mathlib/RingTheory/Bialgebra/Equiv.lean:41]: yeah, it's supposed to be like a combination of the `CoalgEquiv` and `AlgEquiv` notation. But it's hard to guess. Is there any other notation you'd suggest?

Comment (riccardobrasca) [reply Mathlib/RingTheory/Bialgebra/Equiv.lean:41]: Ah, I see, the point is that there are both the `a` and the `c`, so it is bialgebra hom. It makes sense.

**Pull request closed without merging**